src/cosmic_ray/interceptors/pragma_no_mutate.py

```diff
@@ -0,0 +1,47 @@
+"""An interceptor that uses metadata: no mutate to determine when specific mutations
+should be skipped.
+"""
+import re
+from functools import lru_cache
+import logging
+
+from cosmic_ray.work_item import WorkerOutcome, WorkResult
+
+log = logging.getLogger()
+
+
+def intercept(work_db):
+    """Look for WorkItems in `work_db` that should not be mutated due to spor metadata.
+
+    For each WorkItem, find anchors for the item's file/line/columns. If an
+    anchor exists with metadata containing `{mutate: False}` then the WorkItem
+    is marked as SKIPPED.
+    """
+
+    @lru_cache()
+    def file_contents(file_path):
+        "A simple cache of file contents."
+        with file_path.open(mode="rt") as handle:
+            return handle.readlines()
+
+    re_is_mutate = re.compile(r'.*#.*pragma:.*no mutate.*')
```

Comment:
What if this is inside a string literal?

Reply:
I know, I can have false positive. I can lost some mutation.
Have you a best algorihtm ?

Reply:
Is there a package you can use that parses the actual code so that it would understand python syntax? I guess you'd have to parse the entire module to do this correctly.

What you're doing might be state-of-the art...I'm honestly not sure. But I'm very reluctant to make unnecessary guesses.

Reply:
What I'm going to do is accept your PR as-is, and then create a "research issue" to remind us to investigate alternative approaches to detecting this pragma. I think this PR is really good, so I'd like to get it in sooner rather than later.